docs/source/x_ext.rst

```diff
@@ -630,6 +636,47 @@ Whether a load or store is treated as being speculative or not by the |processor
 ignore whatever value it might have communicated via ``commit_kill`` with respect to whether it treats a memory request as speculative or not. A |coprocessor|
 is allowed to signal ``spec`` = 1 without taking the commit transaction into account (so for example even after ``commit_kill`` = 0 has already been signaled).
 
+The ``addr`` signal indicates the (byte) start address of the memory transaction. Transactions on the memory (request/response) interface cannot cross a X_MEM_WIDTH (bus width) boundary.
+The ``be`` signal indicates on what byte lanes to expect valid data for both read and write transactions. ``be[n]`` determines the validity of data bits ``8*N+7:8*N``.
+There are no limitations on the allowed ``be`` values.
+The ``size`` signal indicates the size of the memory transaction. ``size`` shall reflect a naturally aligned range of byte lanes to be used in a transaction.
+The size of a transaction shall not exceed the maximum mememory access width (memory bus width) as determined by ``X_MEM_WIDTH``.
+The ``addr`` signal shall be consistent with the ``be`` signal, i.e. if the maximum memory access width (memory bus width) is 2^N bytes (N=2,3,4,5) and the lowest set bit in
+``be`` is at index IDX, then ``addr[N-1:0]`` shall be at most IDX.
+
+When for example performing a transaction that uses the middle two bytes on a 32-bit wide memory interface, the following (equivalent) `be``, ``size``, ``addr[1:0]`` combinations can be used:
+
+* ``be`` = 4'b0110, ``size`` = 3'b010``, ``addr[1:0]`` = 2'b00.
+* ``be`` = 4'b0110, ``size`` = 3'b010``, ``addr[1:0]`` = 2'b01.
+
+Note that a word transfer is needed in this example because the two bytes transfered are not halfword aligned.
+
+Unaligned (i.e. non naturally aligned) transactions are supported over the memory (request/response) interface using the ``be`` signal. Not all unaligned memory operations
+can however be performed as single transactions on the memory (request/response) interface. Specifically if an unaligned memory operation crosses a X_MEM_WIDTH boundary, then it shall
+be broken into multiple transactions on the memory (request/response) interface by the |coprocessor|.
+
+The ``attr`` signal indicates the attributes of the memory transaction.
+
+``attr[0]`` indicates whether the transaction is a modifiable transaction. This bit shall be set if the
+transaction results from modifications already done in the |coprocessor| (e.g. merging, splitting, or using a transaction size larger than strictly needed (without changing the active byte lanes)) or
+if the |coprocessor| allows such modifications of this transaction at the system level. The |processor| shall check whether a modifiable transaction to the requested
```

Comment:
Do I understand this correctly that `attr[0]` shall be set when the coprocessor allows modifications to a transaction, although it did not perform any modifications itself? How does a coprocessor decide whether it allows modifications or not?

Reply:
Hi @michael-platzer Your understanding is correct, but I agree it actually would not make sense as the coprocessor indeed would have no way to make this decision. I will update/fix the spec.